### DISCUSSION TITLE
Pratica em Js

### DISCUSSION THREAD

**Discussion by giosassis** [frontendbr/forum#2110] [Perguntas e Respostas]
Opa, e aí? Tudo bom?
Há quatro meses estudo javascript, consigo entender a sintaxe completamente mas sou péssima para interpretar algoritmos e resolver exercícios lógicos. Conheço o code wars, mas alguem tem mais sugestão de material, site, exercícios que possa me ajudar a por isso em prática pra eu conseguir destravar???

**Comment by igorlvicente** [comment]
HackerRank e URI Online Judge vão na mesma linha.

**Comment by expalmer** [comment]
Semana que vem se quiser a gente faz uma call e te dou uma mentoria para entender teus objetivos.
Não só você mas quem quiser fica a vontade e me manda uma e-mail, vou começar a dar umas mentorias free pra ajudar a comunidade e quem está começando.

**Reply by giosassis** [reply]
Meu Deus, MUITO OBRIGADA! Posso entrar em contato com você por email??

**Reply by expalmer** [reply]
claro pode sim, expalmer@gmail

**Comment by vitebo** [comment]
eu gosto bastante desse cara: https://www.devchallenge.com.br/
principalmente se voc estiver focando em front-end, pq ele exercita tanto a parte logica do JS quanto a habilidade de codar interface, dependendo do desafio

**Reply by giosassis** [reply]
opa, muito obrigada!!!!!

**Comment by victorferreira** [comment]
Tem também o [Exercism](https://exercism.io/) que, além do exercícios, tem revisores que ajudam bastante com feedbacks sobre o seu código.

**Comment by nazarepiedady** [comment]
Os conhecimentos que se encontram dentro dos livros que passarei a lhe recomendar, são atemporais.

1 - [JavaScript  Guia Definitivo](https://www.amazon.com/JavaScript-Definitivo-Portuguese-David-Flanagan-ebook/dp/B016N7G8EK/ref=sr_1_2?dchild=1&keywords=JavaScript+Guia+Definitivo&qid=1629363961&sr=8-2)
2 - [Coleção Cangaceiro JavaScript](https://www.casadocodigo.com.br/products/colecao-cangaceiro-javascript)
3 - [Estruturas de dados e algoritmos com JavaScript](https://novatec.com.br/livros/estruturas-de-dados-algoritmos-em-javascript-2ed/)
4 - [Lógica de Programação e Algoritmos com JavaScript](https://novatec.com.br/livros/logica-programacao-algoritmos-com-javascript/)

**Reply by giosassis** [reply]
Muito obrigada por compartilhar comigo!

**Comment by MiqueiasGFernandes** [comment]
Recomendo esse repositório: https://github.com/florinpop17/app-ideas

Tem desafio de todos os níveis!

**Reply by giosassis** [reply]
Muito obrigada!
